From 64be676292b62afe240ae19798972097d0d7f87e Mon Sep 17 00:00:00 2001
From: Eric Bernstein <eric.bernstein@amd.com>
Date: Mon, 27 Nov 2017 10:55:52 -0500
Subject: [PATCH 4435/5855] drm/amd/display: use REG_UPDATE for MPC mux

Use REG_UPDATE instead of REG_SET for programming MPC out mux.

Signed-off-by: Eric Bernstein <eric.bernstein@amd.com>
Reviewed-by: Tony Cheng <Tony.Cheng@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/display/dc/dcn10/dcn10_mpc.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/dc/dcn10/dcn10_mpc.c b/drivers/gpu/drm/amd/display/dc/dcn10/dcn10_mpc.c
index 1e72bd4..179890b 100644
--- a/drivers/gpu/drm/amd/display/dc/dcn10/dcn10_mpc.c
+++ b/drivers/gpu/drm/amd/display/dc/dcn10/dcn10_mpc.c
@@ -218,7 +218,7 @@ struct mpcc *mpc1_insert_plane(
 	if (tree->opp_list == insert_above_mpcc) {
 		/* insert the toppest mpcc */
 		tree->opp_list = new_mpcc;
-		REG_SET(MUX[tree->opp_id], 0, MPC_OUT_MUX, mpcc_id);
+		REG_UPDATE(MUX[tree->opp_id], MPC_OUT_MUX, mpcc_id);
 	} else {
 		/* find insert position */
 		struct mpcc *temp_mpcc = tree->opp_list;
@@ -275,11 +275,11 @@ void mpc1_remove_mpcc(
 		if (mpcc_to_remove->mpcc_bot) {
 			/* set the next MPCC in list to be the top MPCC */
 			tree->opp_list = mpcc_to_remove->mpcc_bot;
-			REG_SET(MUX[tree->opp_id], 0, MPC_OUT_MUX, tree->opp_list->mpcc_id);
+			REG_UPDATE(MUX[tree->opp_id], MPC_OUT_MUX, tree->opp_list->mpcc_id);
 		} else {
 			/* there are no other MPCC is list */
 			tree->opp_list = NULL;
-			REG_SET(MUX[tree->opp_id], 0, MPC_OUT_MUX, 0xf);
+			REG_UPDATE(MUX[tree->opp_id], MPC_OUT_MUX, 0xf);
 		}
 	} else {
 		/* find mpcc to remove MPCC list */
@@ -359,7 +359,7 @@ void mpc1_mpc_init(struct mpc *mpc)
 
 	for (opp_id = 0; opp_id < MAX_OPP; opp_id++) {
 		if (REG(MUX[opp_id]))
-			REG_SET(MUX[opp_id], 0, MPC_OUT_MUX, 0xf);
+			REG_UPDATE(MUX[opp_id], MPC_OUT_MUX, 0xf);
 	}
 }
 
-- 
2.7.4

